guardrails/llm_providers.py

```diff
@@ -48,15 +48,18 @@ def __call__(self, *args, **kwargs):
         return result
 
 
-def nonchat_prompt(prompt: str) -> str:
+def nonchat_prompt(prompt: str, instructions: Optional[str] = None, **kwargs) -> str:
     """Prepare final prompt for nonchat engine."""
+    if instructions:
+        prompt = "\n\n".join([instructions, prompt])
```

Comment:
Makes a lot of sense, only counterargument is that then users would be inclined to create different guards for text and chat models, which may be just fine.

One thing to consider if there are some integrations with vector dbs / embedding endpoints, it may be useful to have a part of the prompt to be embedded (maybe the prompt), but the full guard be passed to the model (instructions and prompt). Is it relevant?

Reply:
That seems reasonable.

Re: embedding parts of the prompt -- I think it should be possible to do this by accessing attributes of the `Prompt` class.